lib/ret/storage.ex

```diff
@@ -111,6 +111,18 @@ defmodule Ret.Storage do
     |> Enum.into(%{})
   end
 
+  # Similar to promote above, but allows for passing nil. Useful for optional upload fields
+  def promote_optional(map, %Account{} = account) when is_map(map) do
+    map
+    |> Enum.map(fn
+      {k, {nil, nil}} -> {k, {:ok, nil}}
+      {k, {nil, nil, nil}} -> {k, {:ok, nil}}
+      {k, {id, key}} -> {k, promote(id, key, nil, account)}
+      {k, {id, key, promotion_token}} -> {k, promote(id, key, promotion_token, account)}
```

Comment:
Unrelated to promotion semantics (which I haven't thought enough about to comment on), this and the other `promote` functions might be easier to follow if they changed some of the tuples below to maps and accepted named arguments (maps again) instead of positional arguments.

Reply:
I do think the tuples make sense for the return value since its quite a standard pattern, but agree that the multiple levels of tuples going on here is confusing. Changing is definitely out of scope for this PR, but worth us thinking about.